client/post-editor/media-modal/index.jsx

```diff
@@ -210,13 +210,52 @@ module.exports = React.createClass( {
 		this.setView( ModalViews.IMAGE_EDITOR );
 	},
 
-	onImageEditorSave: function() {
+	onImageExtracted( blob, imageEditorProps ) {
+		const {
+			fileName,
+			site,
+			resetAllImageEditorState
+		} = imageEditorProps;
+
+		const mimeType = MediaUtils.getMimeType( fileName );
+
+		// check if a title is already post-fixed with '(edited copy)'
+		const editedCopyText = this.translate(
+			'%(title)s (edited copy)', {
+				args: {
+					title: ''
+				}
+			} );
+
+		let { title } = imageEditorProps;
+
+		if ( title.indexOf( editedCopyText ) === -1 ) {
+			title = this.translate(
+				'%(title)s (edited copy)', {
+					args: {
+						title: title
+					}
+				} );
+		}
+
+		MediaActions.add( site.ID, {
+			fileName: fileName,
+			fileContents: blob,
+			title: title,
+			mimeType: mimeType
+		} );
+
+		resetAllImageEditorState();
+
 		MediaActions.setLibrarySelectedItems( this.props.site.ID, [] );
 		this.setView( ModalViews.LIST );
 	},
 
-	onImageEditorCancel: function() {
+	onImageEditorCancel: function( imageEditorProps ) {
 		const item = this.props.mediaLibrarySelectedItems[ this.getDetailSelectedIndex() ];
+
+		imageEditorProps.resetAllImageEditorState();
```

Comment:
E.g. image editor in devdocs: before, when I didn't hide the Cancel button, clicking on it would reset the image with its src and would render the image editor unusable in devdocs.